frontend/src/metabase/visualizations/visualizations/Funnel.jsx

```diff
@@ -110,36 +111,53 @@ export default class Funnel extends Component {
       showColumnSetting: true,
       marginBottom: "0.625rem",
     }),
+    "funnel.order_dimension": {
+      getValue: (_series, settings) => settings["funnel.dimension"],
+      readDependencies: ["funnel.rows"],
+    },
     "funnel.rows": {
       section: t`Data`,
       widget: ChartSettingOrderedSimple,
-      isValid: (series, settings) => {
-        const funnelRows = settings["funnel.rows"];
 
-        if (!funnelRows || !_.isArray(funnelRows)) {
-          return false;
-        }
-        if (!funnelRows.every(setting => setting.originalIndex !== undefined)) {
-          return false;
+      getValue: (series, settings) => {
+        const seriesOrder = settings["funnel.rows"];
+        const seriesKeys = series.map(s => keyForSingleSeries(s));
+        const orderDimension = settings["funnel.order_dimension"];
+        const dimension = settings["funnel.dimension"];
+
+        const getDefault = keys =>
+          keys.map(key => ({
+            key,
+            name: key,
+            enabled: true,
+          }));
+        if (
+          !seriesOrder ||
+          !_.isArray(seriesOrder) ||
+          !seriesOrder.every(setting => setting.key !== undefined) ||
+          orderDimension !== dimension
+        ) {
+          return getDefault(seriesKeys);
         }
 
-        return (
-          funnelRows.every(setting => series[setting.originalIndex]) &&
-          funnelRows.length === series.length
-        );
-      },
+        const removeMissingOrder = (keys, order) =>
+          order.filter(o => keys.includes(o.key));
+        const newKeys = (keys, order) =>
+          keys.filter(key => !order.find(o => o.key === key));
 
-      getDefault: transformedSeries => {
-        return transformedSeries.map(s => ({
-          name: s.card.name,
-          originalIndex: s.card.originalIndex,
-          enabled: true,
-        }));
+        return [
+          ...removeMissingOrder(seriesKeys, seriesOrder),
+          ...getDefault(newKeys(seriesKeys, seriesOrder)),
```

Comment:
@adam-james-v We now change the way we reorder funnel charts too, which means the work for both static combo and funnel charts are to be almost the same now. This means [Funnel Static-viz Order not respected fix](https://github.com/metabase/metabase/pull/26596/files#top) isn't applicable with the new way we sort orders.

I can rework static funnel charts too since I'm about to jump on to static combo charts. What do you think?

Reply:
@Kelvin, yea go ahead. I'll close my funnel charts PR and let you take it, as long as you think it's not too much work.

Reply:
It's not too much, but unfortunately, it's blocked due to the change needs some pieces that I created in [Match static combo chart colors and legends with app viz colors](https://github.com/metabase/metabase/pull/26211#top) And also this PR needs to be merged too before I can continue working on reordering static charts.

Reply:
@adam-james-v  It's not too much, but unfortunately, it's blocked due to the change needs some pieces that I created in [Match static combo chart colors and legends with app viz colors](https://github.com/metabase/metabase/pull/26211#top) And also this PR needs to be merged too before I can continue working on reordering static charts.